src/spinnaker_ros_interface_main.cc

```diff
@@ -291,7 +298,12 @@ void CaptureLoop(CameraPtr pCam) {
           net_time_offset = ros::Time::now().toSec() - 1e-9 * static_cast<double>(pResultImage->GetTimeStamp());
         }
         image.header.stamp.fromSec(net_time_offset + 1e-9 * (static_cast<double>(pResultImage->GetTimeStamp())));
-        camera_info_.header.stamp =  image.header.stamp;
+
+        if (CONFIG_ros_pub_camera_info){
+          camera_info_.header.stamp =  image.header.stamp;
+          // Spin to check for and execute set_camera_info_srv_ service requests
+          ros::spinOnce();
```

Comment:
I mainly added the conditional config parameter because i was thinking that ROS spin might be a kinda "heavy" call. If not I think we can do away with the conditional and just make the behavior I added the default behavior.